util.lua

```diff
@@ -63,32 +63,45 @@ digtron.mark_diggable = function(pos, nodes_dug, player)
 	end
 
 	local targetdef = minetest.registered_nodes[target.name]
-	if targetdef == nil or targetdef.can_dig == nil or targetdef.can_dig(pos, player) then
+	if targetdef == nil then
+		minetest.log("warning",
+			string.format("Digtron encountered unknown node %s at (%d, %d, %d), not digging", target.name, pos.x, pos.y, pos.z))
+	elseif (targetdef.diggable or targetdef.buildable_to) and
+		(targetdef.can_dig == nil or targetdef.can_dig(pos, player)) and
+		targetdef.on_dig == minetest.nodedef_default.on_dig and
+		target.name ~= "air" then
```

Comment:
This function becomes somewhat inefficient. `minetest.get_item_group` is used often, and each looks up the node in `minetest.registered_nodes`. Might it be more beneficial to preprocess all `core.registered_nodes` in a `core.register_on_mods_loaded` callback, or `core.after(0, func)`?

i.e.
```Lua
-- key = node name
-- value = digging cost (or 0)
local diggable_nodes = {}

-- in startup callback: add nodes

-- in util.lua
	local material_cost = diggable_nodes[node.name]
	if not material_cost then
		return 0 -- not diggable
	end
	-- add to nodes_dug
	-- get drops, etc.
	return material_cost, drops
```

EDIT: This is just an idea. I might do this in a later PR - unless you'd like to implement it here.